samples/bluetooth/mesh/sensor_server/src/model_handler.c

```diff
@@ -737,46 +922,40 @@ static void button_handler_cb(uint32_t pressed, uint32_t changed)
 					     &present_amb_light_level, &val);
 		if (err) {
 			printk("Error publishing present ambient light level (%d)\n", err);
-		} else {
-			printk("Publishing present ambient light level %s\n",
-			       bt_mesh_sensor_ch_str(&val));
 		}
 	}
-
 	if (BUTTON_PRESSED(pressed, changed, BIT(1))) {
-		int64_t thres_micros;
-		enum bt_mesh_sensor_value_status status;
-
-		status = bt_mesh_sensor_value_to_micro(&pres_mot_thres, &thres_micros);
-		if (bt_mesh_sensor_value_status_is_numeric(status)) {
-			k_work_reschedule(&presence_detected_work, K_MSEC(thres_micros / 10000));
-			printk("Presence will be published in %d ms\n",
-			       (uint16_t)(thres_micros / 10000));
-		} else {
-			/* Value is not known, register presence immediately */
-			k_work_reschedule(&presence_detected_work, K_NO_WAIT);
-		}
+		presence_button_handler(true);
 	}
-
 	if (BUTTON_RELEASED(pressed, changed, BIT(1))) {
-		if (!pres_detect) {
-			printk("Publishing presence is aborted\n");
-			k_work_cancel_delayable(&presence_detected_work);
-		} else {
-			int err;
-			struct bt_mesh_sensor_value val = BOOLEAN_INIT(false);
+		presence_button_handler(false);
+	}
+	if (BUTTON_PRESSED(pressed, changed, BIT(2))) {
+		motion_button_handler(true);
+	}
+	if (BUTTON_RELEASED(pressed, changed, BIT(2))) {
+		motion_button_handler(false);
+	}
+	if (BUTTON_PRESSED(pressed, changed, BIT(3))) {
+		int err;
+		static int people_count_idx;
+		struct bt_mesh_sensor_value val;
 
-			err = bt_mesh_sensor_srv_pub(&occupancy_sensor_srv, NULL,
-						&presence_sensor, &val);
+		dummy_people_count_value = dummy_people_count_values[people_count_idx++];
+		people_count_idx = people_count_idx % ARRAY_SIZE(dummy_people_count_values);
```

Comment:
Suggestion: I prefer `people_count_idx %= ARRAY_SIZE(...);`, I think it's more readable. But up to you.